src/module.c

```diff
@@ -6461,6 +6461,38 @@ const char *RM_CallReplyProto(RedisModuleCallReply *reply, size_t *len) {
     return callReplyGetProto(reply, len);
 }
 
+/* Redis only exposes "side effects" of command execution at the end of an execution unit.
+ *
+ * Side effects include propagating replication/AOF operations and executing the commands of clients blocked on keys
+ * modified by commands
+ *
+ * In general, for modules, execution unit start and end with the taking/release of the GIL.  For module code executed
+ * on the Redis main thread, this is done implicitly by Redis itself, while for module code executed on a separate
+ * thread, this is done explicitly by the module itself.
+ *
+ * This function enables modules to flush the execution unit, thereby having side effects become visible during
+ * execution.
+ *
+ * Examples of usage might be that one is running multiple commands, one after another, with RM_Call and wants
+ * processCommand() like semantics, where unblocked clients will be executed before the next RM_Call is made.
+ *
+ * Note: while one Modules that calls another module via RM_Call won't nest if they are operating on a separate thread,
+ * as each will need to grab the GIL independently, they can nest if both operate on the main thread.  In that case,
+ * it doesn't make sense for the further nested module to be able to cause side effects to the top level module, and
+ * therefore this function only succeeds if running at an execution_nesting level of 1.
+ */
+int RM_FlushExecutionUnit() {
+    if (server.execution_nesting != 1)
+        return REDISMODULE_ERR;
+
+    propagatePendingCommands();
+
+    if (listLength(server.ready_keys))
+        handleClientsBlockedOnKeys();
+
+    return REDISMODULE_OK;
+}
```

Comment:
why don't we add `handleClientsBlockedOnKeys` to `postExecutionUnitOperations` and just call the latter here?

it feels weird to have just a subset of the post-exec-unit operation done...

Reply:
I agree that would be better, just have to understand all the impacts that could have.

Reply:
yes, maybe it requires moving code around, but i think we need to give it a shot